### PR TITLE
fix: is_commutative returned TRUE for non-commutative operators

### PR DESCRIPTION
**Pull request by joske**
This `is_commutative` function had a glaring bug: it returned TRUE for operators like `<` etc (and there were no tests).  This leads to these operators NOT being mutated while they should.

Also it had complicated code to check for function calls in the operands to the operator being evaluated. This leads to false positives like

```rust
-            (found, i + self.inline_vec.length())
+            (found, self.inline_vec.length() + i)
```

```rust
-        self.length() == 0
+        0 == self.length()
```

The exps are the locations of the operands to the op that is potentially mutated as far as I understand it. If any of these functions would have side-effects, then yes, I would agree it could be unsafe to say it's commutative. However, I would say that having MANY false positives is worse than missing a few corner cases that would lead to a live mutant.

This PR also adds unit tests for positive and negative cases.

### CODE REVIEW

File: move-mutator/src/operators/binary_swap.rs [hunk 1]
```diff
@@ -168,4 +146,70 @@ mod tests {
...
+        assert!(operator.is_commutative());
+        let loc = Loc::new(fid, codespan::Span::new(0, 0));
+        let operator = BinarySwap::new(Operation::BitOr, loc, vec![]);
+        assert!(operator.is_commutative());
+        let loc = Loc::new(fid, codespan::Span::new(0, 0));
+        let operator = BinarySwap::new(Operation::BitAnd, loc, vec![]);
+        assert!(operator.is_commutative());
+        let loc = Loc::new(fid, codespan::Span::new(0, 0));
+        let operator = BinarySwap::new(Operation::Xor, loc, vec![]);
+        assert!(operator.is_commutative());
```
[the remark this answers is not in the document]
Comment (joske) [reply move-mutator/src/operators/binary_swap.rs]: can not share the `loc` variable as it would be moved.
